src/libraries/System.IO.FileSystem.Watcher/tests/Args.FileSystemEventArgs.cs

```diff
@@ -8,28 +8,71 @@ namespace System.IO.Tests
     public class FileSystemEventArgsTests
     {
         [Theory]
+        [PlatformSpecific(TestPlatforms.Windows)]
         [InlineData(WatcherChangeTypes.Changed, "C:", "foo.txt")]
+        [InlineData(WatcherChangeTypes.Changed, "D:\\", "foo.txt")]
+        [InlineData(WatcherChangeTypes.Changed, "E:\\bar", "foo.txt")]
         [InlineData(WatcherChangeTypes.All, "C:", "foo.txt")]
-        [InlineData((WatcherChangeTypes)0, "", "")]
-        [InlineData((WatcherChangeTypes)0, "", null)]
+        [InlineData(WatcherChangeTypes.All, "D:\\", "foo.txt")]
+        [InlineData(WatcherChangeTypes.All, "E:\\bar", "foo.txt")]
         public static void FileSystemEventArgs_ctor(WatcherChangeTypes changeType, string directory, string name)
         {
             FileSystemEventArgs args = new FileSystemEventArgs(changeType, directory, name);
 
-            if (!directory.EndsWith(Path.DirectorySeparatorChar.ToString(), StringComparison.Ordinal))
-            {
-                directory += Path.DirectorySeparatorChar;
-            }
+            directory = AppendDirectorySeparator(directory);
```

Comment:
For `C:` we shouldn't append a separator to `directory` if there is not one. Such path is ["A relative path from the current directory of the C: drive."](https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats).

We should rather add another parameter `expectedFullPath` and assert that against `args.FullPath`. Thoughts?

Reply:
Agree. Added in next commit.